### PR TITLE
Delete VM extensions manually before deleting the whole resource

### PR DESCRIPTION
**Pull request by tanya-borisova**
# PR for issue https://github.com/microsoft/AzureTRE/issues/2138

## What is being addressed

VM extensions need to be deleted manually before deleting the resource for the VM, as a workaround for a known bug: https://github.com/hashicorp/terraform-provider-azurerm/issues/6098 

## How is this addressed

- Added a script to get from terraform state and delete all VM extensions

### CODE REVIEW

File: templates/workspace_services/guacamole/user_resources/guacamole-azure-linuxvm/delete_vm_extensions.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+set -o errexit
+set -o pipefail
+# Uncomment this line to see each command for debugging (careful: this will show secrets!)
+set -o xtrace
```
Comment (ross-p-smith) [inline templates/workspace_services/guacamole/user_resources/guacamole-azure-linuxvm/delete_vm_extensions.sh:6]: Should this be commented out?

File: templates/workspace_services/guacamole/user_resources/guacamole-azure-windowsvm/delete_vm_extensions.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+set -o errexit
+set -o pipefail
+# Uncomment this line to see each command for debugging (careful: this will show secrets!)
+set -o xtrace
```
Comment (ross-p-smith) [inline templates/workspace_services/guacamole/user_resources/guacamole-azure-windowsvm/delete_vm_extensions.sh:6]: comment this out

File: templates/workspace_services/guacamole/user_resources/guacamole-azure-windowsvm/delete_vm_extensions.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+set -o errexit
```
Comment (ross-p-smith) [inline templates/workspace_services/guacamole/user_resources/guacamole-azure-windowsvm/delete_vm_extensions.sh:3]: Add nounset

**Review by ross-p-smith — APPROVED** [review]

File: templates/workspace_services/guacamole/user_resources/guacamole-azure-linuxvm/delete_vm_extensions.sh (hunk 1 above)
Comment (tanya-borisova) [reply templates/workspace_services/guacamole/user_resources/guacamole-azure-linuxvm/delete_vm_extensions.sh:6]: Forgot! Thanks

File: templates/workspace_services/guacamole/user_resources/guacamole-azure-linuxvm/delete_vm_extensions.sh [hunk 4]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+set -o errexit
+set -o pipefail
```
[the remark this answers is not in the document]
Comment (tanya-borisova) [reply templates/workspace_services/guacamole/user_resources/guacamole-azure-linuxvm/delete_vm_extensions.sh:4]: Good point

**Pull request merged**